core/src/main/scala/sttp/client/SttpApi.scala

```diff
@@ -209,4 +208,47 @@ trait SttpApi extends SttpExtensions with UriInterpolator {
     */
   def multipart[B: BodySerializer](name: String, b: B): Part[BasicRequestBody] =
     Part(name, implicitly[BodySerializer[B]].apply(b), contentType = Some(MediaType.ApplicationXWwwFormUrlencoded))
+
+  /**
+    * Allows to provide additional mappings for error cases based on statusCodes
+    * @param onSuccess transformation which will be applied if the response is successful (2xx)
+    * @param onError transformation which will be applied if the response is unsuccessful (non 2xx)
+    * @tparam DE underlying type of deserialization failure
+    * @tparam HE type which represents known error cases
+    * @tparam T type of the successful response
+    */
+  def asTypedError[DE, HE, T](onSuccess: ResponseAs[Either[DeserializationError[DE], T], Nothing])(
```

Comment:
as this is more of a decorator, maybe a name like `withTypedErrors` would be better?